### DISCUSSION TITLE
How to add a video to a AR scene?

### DISCUSSION THREAD

**Discussion by sarathprasad** [sceneview/sceneview#500] [General]
I'm new to ARcore and SceneView, and I'm struggling to add a video to a model node within an ARScene. Can someone provide guidance or point me to a sample code that demonstrates how to achieve this? Any assistance or example code would be greatly appreciated. Thanks in advance!

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, use `VideoNode`:

```kotlin
ARSceneView {
    AnchorNode(anchor = anchor) {
        VideoNode(
            materialLoader = materialLoader,
            mediaPlayer = mediaPlayer,
            width = 2.0f,
            height = 1.5f
        )
    }
}
```

Create a standard Android `MediaPlayer` for your video source and pass it to `VideoNode`. It renders the video as an `ExternalTexture` on a quad in the AR scene.
